lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -173,17 +171,150 @@ export class BrowserCacheManager extends CacheManager {
         }
     }
 
+    /**
+     * Helper to setItem in browser storage, with cleanup in case of quota errors
+     * @param key
+     * @param value
+     */
+    setItem(key: string, value: string, correlationId: string): void {
+        let tokenKeys: TokenKeys | null = null;
+        let trySetItem = true;
+        while (trySetItem) {
```

Comment:
I'd likely add some endless loop protection.  Let's not do this more than say 10x.  We should have some observability in here.  Both in the out of space case, as well as the fail to remove enough items to insert this into the cache case.

Reply:
I see the removal observability handled below, just not the failure to insert even after we make room.

Reply:
- The endless loop protection is the accessToken key array. This won't run any more times than there are ATs to remove. 
- Observability for "out of space case but we're able to recover" is below - we track the number of ATs we removed. 
- Observability for "failed to remove enough items to insert" is the cacheQuotaExceeded error which will be thrown and logged

Reply:
I feel protecting against future changes is, above all else, why we write tests. Nonetheless I agree the exit conditions here aren't immediately obvious and may cause alarm on first glance. So in the interest of clarity and peace of mind I've refactored into a for loop with max retry of 20. If we're still failing after removing 20 it's unlikely removing the 21st will make a difference anyway. Thanks for the feedback!